src/redisai.h

```diff
@@ -41,7 +41,7 @@ long long MODULE_API_FUNC(RedisAI_TensorDim)(RAI_Tensor* t, int dim);
 size_t MODULE_API_FUNC(RedisAI_TensorByteSize)(RAI_Tensor* t);
 char* MODULE_API_FUNC(RedisAI_TensorData)(RAI_Tensor* t);
 
-RAI_Model* MODULE_API_FUNC(RedisAI_ModelCreate)(int backend, int device,
+RAI_Model* MODULE_API_FUNC(RedisAI_ModelCreate)(RAI_Backend backend, RAI_Device device,
```

Comment:
RAI_Backend and RAI_Device are not define in this h file, other modules will not be able to compile with it. This is why it was int and we have those defines to know which value to pass:
```
#define REDISAI_BACKEND_TENSORFLOW 0
#define REDISAI_BACKEND_TORCH 1
#define REDISAI_BACKEND_ONNXRUNTIME 2

#define REDISAI_DEVICE_CPU 0
#define REDISAI_DEVICE_GPU 1
```

Reply:
The thing is that since now we use function pointers, using `int` here causes a warning because the function signature is not the same as its definition.
So we either switch to `int` or expose the `enums` at the `redisai.h` level.
Which one do you think is the best option?

Reply:
I am not sure why it throw warning, is it here:
```
RedisAI_ ## name = RAI_ ## name;
```

Reply:
So I just compiled it, its this like that cause the problem, I think you can just remove this line since you use directly the RAI_ functions and not the function pointers.

Reply:
Sounds good, thanks!